lib/uktt/http.rb

```diff
@@ -16,26 +20,23 @@ def initialize(connection, options)
     end
 
     def retrieve(resource, query_config = {})
-      resource = File.join(service, 'api', version, resource) if public?
-      resource = File.join('/', resource)
-      resource = "#{resource}#{query_params(query_config)}"
-
-      response = Retriable.retriable(intervals: retriable_intervals) do
-        do_fetch(resource)
-      end
+      resource = File.join(host, 'api', version, resource) if public?
+      response = do_fetch(resource, query_config)
 
       Parser.new(response.body, format).parse
     end
 
-    def self.build(host, version, format, public_routes, retriable_intervals = nil)
-      uri = URI(host)
-      connection = Net::HTTP.new(uri.host, uri.port)
-      connection.use_ssl = uri.scheme.include?('https')
+    def self.build(host, version, format, public_routes, retriable_options = nil)
+      connection = Faraday.new(url: host) do |faraday|
+        faraday.use FaradayMiddleware::FollowRedirects
+        faraday.adapter :net_http_persistent
+        faraday.response :logger if ENV['DEBUG_REQUESTS']
```

Comment:
Is it worth sticking the `RaiseError` middleware in there?

Reply:
I've actually not used it before. I will have a gander

Reply:
Added